internal_task_handlers.go

```diff
@@ -308,83 +324,242 @@ func newWorkflowTaskHandler(
 ) WorkflowTaskHandler {
 	ensureRequiredParams(&params)
 	return &workflowTaskHandlerImpl{
-		domain:                domain,
-		logger:                params.Logger,
-		ppMgr:                 ppMgr,
-		metricsScope:          params.MetricsScope,
-		identity:              params.Identity,
-		enableLoggingInReplay: params.EnableLoggingInReplay,
-		hostEnv:               hostEnv,
+		domain:                 domain,
+		logger:                 params.Logger,
+		ppMgr:                  ppMgr,
+		metricsScope:           params.MetricsScope,
+		identity:               params.Identity,
+		enableLoggingInReplay:  params.EnableLoggingInReplay,
+		disableStickyExecution: params.DisableStickyExecution,
+		hostEnv:                hostEnv,
 	}
 }
 
-// ProcessWorkflowTask processes each all the events of the workflow task.
-func (wth *workflowTaskHandlerImpl) ProcessWorkflowTask(
-	task *s.PollForDecisionTaskResponse,
-	getHistoryPage GetHistoryPage,
-	emitStack bool,
-) (result interface{}, stackTrace string, err error) {
-	if task == nil {
-		return nil, "", errors.New("nil workflowtask provided")
+// TODO: need a better eviction policy based on memory usage
+var workflowCache = cache.New(1000, &cache.Options{
```

Comment:
Shouldn't cache be per worker?

Reply:
i think it should be per process, or we could have multiple caches which could make it hard if we ever want to do something based on memory stats (which is global view).